publication.go

```diff
@@ -18,6 +18,21 @@ import (
 	"go.aporeto.io/elemental"
 )
 
+// ResponseMode represents the response that is expected to be produced by the subscriber
+// handling a publication.
+type ResponseMode int
+
+const (
+	// ReplyWithACK indicates that the subscriber should reply back with an ACK
+	// as soon as it has received the publication BEFORE it starts processing the
+	// publication.
+	ReplyWithACK ResponseMode = iota + 1
```

Comment:
What's with the +1? 

Given that the default value of int's is 0, all `Publications` that are initialized would default to the first value in the iota (even if the client did not specify this field explicitly). By starting at 1, a consumer would be making an explicit decision of their desired response mode.

Reply:
How do we sauy we don't need reply or ack? (Whicj should be 0)

Reply:
We usually call things this way

Type ThingMode int

Const ThingModeA
Const ThingModeB

This makes it easier to find them in autocompletion in general and they are close in documentation

Reply:
> How do we sauy we don't need reply or ack? (Whicj should be 0)

If you don't supply a `PubSubOptPublish` option for setting your desired request mode when calling `Publish`, the default value will be 0 which does not match any of our defined request modes.

As a result, this will match the the default case [here](https://github.com/aporeto-inc/bahamut/pull/59/files#diff-1abbda2347372fced407ce97f82bf317R116) which will simply publish a publication without expecting any response.

Reply:
I see. It would be nice to still have a const for the 0 value, just for clarity